vm-virtio/src/lib.rs

```diff
@@ -117,9 +120,20 @@ const INTERRUPT_STATUS_CONFIG_CHANGED: u32 = 0x2;
 pub enum ActivateError {
     EpollCtl(std::io::Error),
     BadActivate,
-
+    /// Queue number is not correct
+    BadQueueNum,
+    /// Failed to clone Kill event
+    CloneKillEventFd,
+    /// Failed to create Vhost-user interrupt eventfd
+    VhostIrqCreate,
     /// Failed to setup vhost-user daemon.
     VhostUserSetup(fs::Error),
+    /// Failed to setup vhost-user daemon.
+    VhostUserNetSetup(vhost_user::Error),
+    /// Set vring enable failed.
+    VhostUserSetVringEnable(VhostError),
+    /// Set features failed.
+    VhostUserNetSetFeatures(VhostError),
```

Comment:
Will that build when the `vhost-user-master` feature is disabled?

Reply:
No, if neither vhost-user-master nor vhost-user-slave is enabled, these two messages should not be built, they are wrapper of VhostUserProtocol(vhost_user::Error) in vhost_rs::Error, which has attribute #[cfg(any(feature = "vhost-user-master", feature = "vhost-user-slave"))].

Reply:
Yes, it's a good question, I'm considering if we need to have vhost-user devices as features for vm-virtio, and add feature config for the cases like the messages you mentioned above, also like vsock config in firecracker.

Reply:
The error types are vhost_rs::Error. But it doesn't matter, they are removed from lib.rs, new function added in vu_common_ctrl to handle the logic.